spec/policy/soap/policy_spec.lua

```diff
@@ -0,0 +1,128 @@
+local Usage = require('apicast.usage')
+
+describe('policy', function()
+  describe('.rewrite', function()
+    local context -- Context shared between policies
+
+    -- Define a config with 2 rules. One increases hits by 10 and the other by
+    -- 20. Their patterns have values that allow us to easily associate them
+    -- with a SOAP action receive via SOAPAction header or via Content-Type.
+    local policy_config = {
+      mapping_rules = {
+        {
+          pattern = '/soap_action$',
```

Comment:
Because the RFC says these should be full URL we should test also with full URLs. To verify all the escaping works.